app/lib/screens/HomeScreens/ChatScreen.dart

```diff
@@ -0,0 +1,435 @@
+// ignore_for_file: prefer_const_constructors, sized_box_for_whitespace, prefer_final_fields, prefer_typing_uninitialized_variables
+
+import 'dart:typed_data';
+
+import 'package:effektio/common/store/Colors.dart';
+import 'package:effektio/common/store/chatTheme.dart';
+import 'package:effektio/common/widget/AppCommon.dart';
+import 'package:effektio/common/widget/emptyMessagesPlaceholder.dart';
+import 'package:effektio/repository/client.dart';
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+import 'package:effektio_flutter_sdk/effektio_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart';
+import 'package:flutter_chat_ui/flutter_chat_ui.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:open_file/open_file.dart';
+
+class ChatScreen extends StatefulWidget {
+  final Conversation room;
+  final String? user;
+  const ChatScreen({Key? key, required this.room, required this.user})
+      : super(key: key);
+
+  @override
+  State<ChatScreen> createState() => _ChatScreenState();
+}
+
+class _ChatScreenState extends State<ChatScreen> {
+  List<types.Message> _messages = [];
+  late final _user;
+  bool isLoading = false;
+  //incremental, don't know if it has any use but written here as in docs
+  int _page = 0;
+  //starting from this value to show (messages-10) to avoid duplication
+  int _count = 20;
+  @override
+  void initState() {
+    _user = types.User(
+      id: widget.user!,
+      firstName: getNameFromId(widget.user!),
+    );
+    _getMessages();
+    _handleEndReached();
+    _updateState();
+    super.initState();
+  }
+
+  //will detect if any new event is arrived and will re-render the screen
+  void _updateState() async {
+    var timeline = await widget.room.timeline();
+    await timeline.next();
+    var newEvent = await widget.room.latestMessage();
```

Comment:
`timeline.next()` already returns the new event.
```suggestion
    var newEvent = await timeline.next();
```